### PR TITLE
`azurerm_storage_account_network_rules` - split create update and remove ConfigModeAttr / O+C

### PR DESCRIPTION
**Pull request by jackofallops**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

split create/update
remove O+C where possible
remove redundant ConfigModeAttr schema settings

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [ ] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [ ] I have written new tests for my resource or datasource changes & updated any relevant documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_storage_account_network_rules` - split create update and remove ConfigModeAttr / O+C

### CODE REVIEW

**Comment by jackofallops** [comment]
Test looking good:

image

**Review by manicminer — APPROVED** [review]
Thanks @jackofallops, LGTM 👍

File: internal/services/storage/storage_account_network_rules_resource.go [hunk 1]
```diff
@@ -124,7 +119,7 @@ func resourceStorageAccountNetworkRules() *pluginsdk.Resource {
 	}
 }
 
-func resourceStorageAccountNetworkRulesCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+func resourceStorageAccountNetworkRulesCreate(d *pluginsdk.ResourceData, meta interface{}) error {
 	tenantId := meta.(*clients.Client).Account.TenantId
 	client := meta.(*clients.Client).Storage.ResourceManager.StorageAccounts
 	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
```
Comment (stephybun) [inline internal/services/storage/storage_account_network_rules_resource.go:125]:
```suggestion
	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
```

File: internal/services/storage/storage_account_network_rules_resource.go [hunk 2]
```diff
@@ -184,12 +177,74 @@ func resourceStorageAccountNetworkRulesCreateUpdate(d *pluginsdk.ResourceData, m
...
-		d.SetId(id.ID())
+	d.SetId(id.ID())
+
+	return resourceStorageAccountNetworkRulesRead(d, meta)
+}
+
+func resourceStorageAccountNetworkRulesUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	tenantId := meta.(*clients.Client).Account.TenantId
+	client := meta.(*clients.Client).Storage.ResourceManager.StorageAccounts
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
```
Comment (stephybun) [inline internal/services/storage/storage_account_network_rules_resource.go:192]:
```suggestion
	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
```

File: internal/services/storage/storage_account_network_rules_resource.go [hunk 3]
```diff
@@ -184,12 +177,74 @@ func resourceStorageAccountNetworkRulesCreateUpdate(d *pluginsdk.ResourceData, m
 			NetworkAcls: acls,
 		},
 	}
-	if _, err := client.Update(ctx, *id, payload); err != nil {
+	if _, err = client.Update(ctx, *id, payload); err != nil {
 		return fmt.Errorf("updating Network Rules for %s: %+v", *id, err)
```
Comment (stephybun) [inline internal/services/storage/storage_account_network_rules_resource.go:181]: Although it's an update should the error message say creating?
```suggestion
		return fmt.Errorf("creating Network Rules for %s: %+v", *id, err)
```

**Review by stephybun — APPROVED** [review]
Thanks @jackofallops, a few minor comments otherwise LGTM 👍

**Pull request merged**